### DISCUSSION TITLE
Disable pin for one image

### DISCUSSION THREAD

**Discussion by adobe-ddavidso** [renovatebot/renovate#20614] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.148.0

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

3.6.7

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We have several docker images in our project, but one of the docker images we would like to not pin the digest.  We were able to disable the digest updates using this package rule
```
    {
      "matchPackageNames": [
        "docker-asr-release.dr.corp.adobe.com/asr/nodejs_v18"
      ],
      "digest": {
        "enabled": false
      }
    }
```
But the docker image also has a version-like tag, so renovate correctly updates to the new version, but when the version is updated the digest is included.

### What you tried so far.

We attempted to update the package rule to
```
    {
      "matchPackageNames": [
        "docker-asr-release.dr.corp.adobe.com/asr/nodejs_v18"
      ],
      "digest": {
        "enabled": false
      },
      "docker": {
        "pinDigests": false
      }
    }
```
but we got the error `The “docker” object can only be configured at the top level of a config but was found inside “packageRules[2]`

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
You can't nest objects like `digest` and `docker` within `packageRules`

**Reply by adobe-ddavidso** [reply]
Do you have a suggestion on how we would config renovate to never pin a specific image?

**Reply by rarkins** [reply]
In a package rule, use matchPackageNames=x and matchUpdateTypes=digest and enabled=false

**Reply by adobe-ddavidso** [reply]
We added this package rule
```
{
      "matchPackageNames": [
        "docker-asr-release.dr.corp.adobe.com/asr/nodejs_v18"
      ],
      "matchUpdateTypes": ["digest", "pinDigest"],
      "enabled": false
}
```
but renovate still pinned the digest after a `patch` update

![Screen Shot 2023-03-02 at 10 28 54 AM](https://user-images.githubusercontent.com/121135158/222519296-de9e18d4-a61b-4ae5-bfab-ff910891353b.png)

Any other adjustments to the renovate config we might have missed?

**Reply by rarkins** [reply]
You need to set pinDigests=false for that dependency

**Reply by adobe-ddavidso** [reply]
So update the package rule to?
```
{
      "matchPackageNames": [
        "docker-asr-release.dr.corp.adobe.com/asr/nodejs_v18"
      ],
      "matchUpdateTypes": ["digest", "pinDigest"],
      "enabled": false,
      "pinDigests": false
}
```

**Reply by rarkins** [reply]
Just:

```
{
      "matchPackageNames": [
        "docker-asr-release.dr.corp.adobe.com/asr/nodejs_v18"
      ],
      "pinDigests": false
}
```

**Reply by adobe-ddavidso** [reply]
Great!  Thanks!
